diff --git a/src/main/java/com/github/manolo8/darkbot/config/Config.java b/src/main/java/com/github/manolo8/darkbot/config/Config.java
index ae05ee05..5de41af1 100644
--- a/src/main/java/com/github/manolo8/darkbot/config/Config.java
+++ b/src/main/java/com/github/manolo8/darkbot/config/Config.java
@@ -291,7 +291,7 @@ public class Config implements eu.darkbot.api.config.legacy.Config {
 
         public @Option @Visibility(Level.INTERMEDIATE) Other OTHER = new Other();
         public static class Other {
-            public @Option boolean SHOW_INSTRUCTIONS = true;
+            public @Option boolean SHOW_INSTRUCTIONS = false;
             public @Option boolean DISABLE_MASTER_PASSWORD = false;
             public @Option boolean ALWAYS_SHOW_CAPTCHA = false;
             public @Option @Number(min = 10, max = 300) int ZONE_RESOLUTION = 30;
