diff --git a/src/main/java/com/github/manolo8/darkbot/config/Config.java b/src/main/java/com/github/manolo8/darkbot/config/Config.java
index 6cf20b58..5de41af1 100644
--- a/src/main/java/com/github/manolo8/darkbot/config/Config.java
+++ b/src/main/java/com/github/manolo8/darkbot/config/Config.java
@@ -291,6 +291,7 @@ public class Config implements eu.darkbot.api.config.legacy.Config {
 
         public @Option @Visibility(Level.INTERMEDIATE) Other OTHER = new Other();
         public static class Other {
+            public @Option boolean SHOW_INSTRUCTIONS = false;
             public @Option boolean DISABLE_MASTER_PASSWORD = false;
             public @Option boolean ALWAYS_SHOW_CAPTCHA = false;
             public @Option @Number(min = 10, max = 300) int ZONE_RESOLUTION = 30;
diff --git a/src/main/java/com/github/manolo8/darkbot/extensions/features/decorators/InstructionProviderDecorator.java b/src/main/java/com/github/manolo8/darkbot/extensions/features/decorators/InstructionProviderDecorator.java
index d9649b93..c97d948e 100644
--- a/src/main/java/com/github/manolo8/darkbot/extensions/features/decorators/InstructionProviderDecorator.java
+++ b/src/main/java/com/github/manolo8/darkbot/extensions/features/decorators/InstructionProviderDecorator.java
@@ -1,5 +1,6 @@
 package com.github.manolo8.darkbot.extensions.features.decorators;
 
+import com.github.manolo8.darkbot.config.ConfigEntity;
 import com.github.manolo8.darkbot.extensions.features.FeatureDefinition;
 import com.github.manolo8.darkbot.gui.utils.Popups;
 import com.github.manolo8.darkbot.utils.I18n;
@@ -13,7 +14,10 @@ public class InstructionProviderDecorator extends FeatureDecorator<InstructionPr
 
     @Override
     protected void load(FeatureDefinition<InstructionProvider> fd, InstructionProvider obj) {
-        if (obj instanceof Module && !(obj instanceof TemporalModule)) showInstructions(obj, fd.getName());
+        if (obj instanceof Module
+                && !(obj instanceof TemporalModule)
+                && ConfigEntity.INSTANCE.getConfig().BOT_SETTINGS.OTHER.SHOW_INSTRUCTIONS
+        ) showInstructions(obj, fd.getName());
     }
 
     @Override
diff --git a/src/main/resources/lang/strings_en.properties b/src/main/resources/lang/strings_en.properties
index b6ed3d1d..143ec952 100644
--- a/src/main/resources/lang/strings_en.properties
+++ b/src/main/resources/lang/strings_en.properties
@@ -311,6 +311,8 @@ config.bot_settings.performance.always_enabled=Always enabled
 config.bot_settings.other=Other
 config.bot_settings.other.disable_master_password=Disable master password
 config.bot_settings.other.disable_master_password.desc=Enable before having added any logins, or having an empty master password
+config.bot_settings.other.exclusive_gui_showing=Enable exclusive HUD showing
+config.bot_settings.other.exclusive_gui_showing.desc=The bot will show only one HUD window, switch attempt will be suppressed for 10 seconds after the last activity with the active window.
 config.bot_settings.other.always_show_captcha=Always show captcha window
 config.bot_settings.other.always_show_captcha.desc=Always show captcha window, even if any other app is in fullscreen mode. Otherwise, will be shown only on first login or if bot gui has focus
 config.bot_settings.other.zone_resolution=Zone precision
@@ -318,6 +320,8 @@ config.bot_settings.other.zone_resolution.desc=Amount of map subdivisions when s
 config.bot_settings.other.min_tick=Minimum tick time
 config.bot_settings.other.dev_stuff=Developer stuff shown
 config.bot_settings.other.dev_stuff.desc=Enabling this WILL make your bot use more cpu.
+config.bot_settings.other.show_instructions=Show instructions
+config.bot_settings.other.show_instructions.desc=Enabling showing instructions for modules after loading it
 
 # Misc
 misc.editor.checkbox_list.selected={0} selected
