diff --git a/impl/src/main/java/eu/darkbot/impl/managers/EventBroker.java b/impl/src/main/java/eu/darkbot/impl/managers/EventBroker.java
index 2db90ff..6fe48e2 100644
--- a/impl/src/main/java/eu/darkbot/impl/managers/EventBroker.java
+++ b/impl/src/main/java/eu/darkbot/impl/managers/EventBroker.java
@@ -1,5 +1,6 @@
 package eu.darkbot.impl.managers;

+import com.github.m9w.customeventbroker.CustomEventBroker;
 import eu.darkbot.api.events.Event;
 import eu.darkbot.api.events.EventHandler;
 import eu.darkbot.api.events.Listener;
@@ -41,6 +42,7 @@ public class EventBroker implements EventBrokerAPI {

     @Override
     public synchronized void registerListener(@NotNull Listener listener) {
+        CustomEventBroker.INSTANCE.registerListener(listener);
         if (dispatchers.containsKey(listener)) unregisterListener(listener);
         if (eventsBeingSent == 0) {
             dispatchers.put(listener, new EventDispatcher(listener.getClass()));
@@ -51,6 +53,7 @@ public class EventBroker implements EventBrokerAPI {

     @Override
     public synchronized void unregisterListener(@NotNull Listener listener) {
+        CustomEventBroker.INSTANCE.unregisterListener(listener);
         if (eventsBeingSent == 0) {
             dispatchers.remove(listener);
         } else {
