diff --git a/src/main/java/com/github/manolo8/darkbot/Bot.java b/src/main/java/com/github/manolo8/darkbot/Bot.java
index b8d822b6..be393d57 100644
--- a/src/main/java/com/github/manolo8/darkbot/Bot.java
+++ b/src/main/java/com/github/manolo8/darkbot/Bot.java
@@ -23,7 +23,7 @@ import java.security.ProtectionDomain;
 public class Bot {

     public static void main(String[] args) throws IOException {
-        // You can enable hardware acceleration via adding jvm arg: -Dsun.java2d.opengl=True
+        com.github.m9w.hooks.HooksHandler.INSTANCE.beforeMain(args);// You can enable hardware acceleration via adding jvm arg: -Dsun.java2d.opengl=True
         try {
             setupSecurityPolicy();
         } catch (Exception e) {
diff --git a/src/main/java/com/github/manolo8/darkbot/Main.java b/src/main/java/com/github/manolo8/darkbot/Main.java
index 71fd9978..fc8cff57 100644
--- a/src/main/java/com/github/manolo8/darkbot/Main.java
+++ b/src/main/java/com/github/manolo8/darkbot/Main.java
@@ -146,7 +146,7 @@ public class Main extends Thread implements PluginListener, BotAPI {
         VerifierChecker.getAuthApi().setupAuth();
         this.pluginAPI.addInstance(VerifierChecker.getAuthApi());

-        this.starManager     = pluginAPI.requireInstance(StarManager.class);
+        com.github.m9w.hooks.HooksHandler.INSTANCE.beforeApiInit(); this.starManager     = pluginAPI.requireInstance(StarManager.class);
         this.mapManager      = pluginAPI.requireInstance(MapManager.class);
         this.settingsManager = pluginAPI.requireInstance(SettingsManager.class);
         this.facadeManager   = pluginAPI.requireInstance(FacadeManager.class);
@@ -161,7 +161,7 @@ public class Main extends Thread implements PluginListener, BotAPI {
         this.featureRegistry = pluginAPI.requireInstance(FeatureRegistry.class);
         this.repairManager   = pluginAPI.requireInstance(RepairManager.class);
         this.botInstaller = pluginAPI.requireInstance(BotInstaller.class);
-        this.eventBroker = pluginAPI.requireAPI(EventBrokerAPI.class);
+        this.eventBroker = pluginAPI.requireAPI(EventBrokerAPI.class); com.github.m9w.hooks.HooksHandler.INSTANCE.afterApiInit();

         API = configManager.getAPI(pluginAPI);
         API.setSize(config.BOT_SETTINGS.API_CONFIG.width, config.BOT_SETTINGS.API_CONFIG.height);
@@ -235,7 +235,7 @@ public class Main extends Thread implements PluginListener, BotAPI {

         this.form.tick();
         this.configManager.saveChangedConfig();
-        this.configChange.tick();
+        this.configChange.tick(); com.github.m9w.hooks.HooksHandler.INSTANCE.eachTick();

         processTasks();
     }
