diff --git a/src/main/java/com/github/manolo8/darkbot/Bot.java b/src/main/java/com/github/manolo8/darkbot/Bot.java
index b8d822b6..854710fb 100644
--- a/src/main/java/com/github/manolo8/darkbot/Bot.java
+++ b/src/main/java/com/github/manolo8/darkbot/Bot.java
@@ -3,6 +3,7 @@ package com.github.manolo8.darkbot;
 import com.formdev.flatlaf.FlatDarkLaf;
 import com.formdev.flatlaf.ui.FlatNativeWindowBorder;
 import com.formdev.flatlaf.util.SystemInfo;
+import com.github.m9w.hooks.HooksHandler;
 import com.github.manolo8.darkbot.extensions.plugins.PluginClassLoader;
 import com.github.manolo8.darkbot.gui.MainGui;
 import com.github.manolo8.darkbot.utils.LibSetup;
@@ -23,6 +24,7 @@ import java.security.ProtectionDomain;
 public class Bot {

     public static void main(String[] args) throws IOException {
+        HooksHandler.INSTANCE.beforeMain(args);
         // You can enable hardware acceleration via adding jvm arg: -Dsun.java2d.opengl=True
         try {
             setupSecurityPolicy();
diff --git a/src/main/java/com/github/manolo8/darkbot/Main.java b/src/main/java/com/github/manolo8/darkbot/Main.java
index 71fd9978..abe25df3 100644
--- a/src/main/java/com/github/manolo8/darkbot/Main.java
+++ b/src/main/java/com/github/manolo8/darkbot/Main.java
@@ -1,5 +1,6 @@
 package com.github.manolo8.darkbot;

+import com.github.m9w.hooks.HooksHandler;
 import com.github.manolo8.darkbot.backpage.BackpageManager;
 import com.github.manolo8.darkbot.config.Config;
 import com.github.manolo8.darkbot.config.ConfigHandler;
@@ -146,6 +147,7 @@ public class Main extends Thread implements PluginListener, BotAPI {
         VerifierChecker.getAuthApi().setupAuth();
         this.pluginAPI.addInstance(VerifierChecker.getAuthApi());

+        HooksHandler.INSTANCE.beforeApiInit();
         this.starManager     = pluginAPI.requireInstance(StarManager.class);
         this.mapManager      = pluginAPI.requireInstance(MapManager.class);
         this.settingsManager = pluginAPI.requireInstance(SettingsManager.class);
@@ -162,6 +164,7 @@ public class Main extends Thread implements PluginListener, BotAPI {
         this.repairManager   = pluginAPI.requireInstance(RepairManager.class);
         this.botInstaller = pluginAPI.requireInstance(BotInstaller.class);
         this.eventBroker = pluginAPI.requireAPI(EventBrokerAPI.class);
+        HooksHandler.INSTANCE.afterApiInit();
 
         API = configManager.getAPI(pluginAPI);
         API.setSize(config.BOT_SETTINGS.API_CONFIG.width, config.BOT_SETTINGS.API_CONFIG.height);
